kyo-core/shared/src/main/scala/kyo/Flat.scala

```diff
@@ -2,19 +2,10 @@ package kyo
 
 import internal.FlatImplicits
 
-sealed trait Flat[-T]
+opaque type Flat[T] = Null
```

Comment:
This encoding saves a few bytecodes in `core`.

Reply:
I also tried using the new `erased` modifier but the build times increased too much.

Reply:
How about `AnyVal`? I don't love null, given that it may be difficult to debug if any methods are invoked on the bypass instance.

Reply:
It could be an issue if someone calls `flat.toString` for example but that wouldn't make much sense. Ideally, this type should have no runtime representation with `erased`. Using `Null` here reduces the core size because it gets inlined as a `null` constant. I had used `Unit` initially but just pointing to an actual object requires more instructions.